server/src/main/java/org/opensearch/search/query/ConcurrentQueryPhaseSearcher.java

```diff
@@ -77,19 +76,18 @@ private static boolean searchWithCollectorManager(
             collectorManager = QueryCollectorManagerContext.createQueryCollectorManager(collectorContexts);
         }
 
-        try {
-            final ReduceableSearchResult result = searcher.search(query, collectorManager);
-            result.reduce(queryResult);
-        } catch (EarlyTerminatingCollector.EarlyTerminationException e) {
-            queryResult.terminatedEarly(true);
-        } catch (TimeExceededException e) {
+        final ReduceableSearchResult result = searcher.search(query, collectorManager);
+        result.reduce(queryResult);
+        if (searchContext.isSearchTimedOut()) {
             assert timeoutSet : "TimeExceededException thrown even though timeout wasn't set";
             if (searchContext.request().allowPartialSearchResults() == false) {
-                // Can't rethrow TimeExceededException because not serializable
                 throw new QueryPhaseExecutionException(searchContext.shardTarget(), "Time exceeded");
             }
             queryResult.searchTimedOut(true);
         }
+        if (searchContext.isTerminatedEarly()) {
+            queryResult.terminatedEarly(true);
```

Comment:
this will not be needed without support for `forceTermination` as it is handled via the `EarlyTerminatingCollector::reduce`